libnmstate/nm/wired.py

```diff
@@ -0,0 +1,61 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate import nmclient
+
+
+def create_setting(iface_state, base_con_profile):
```

Comment:
I would prefer to see this function do one thing and not two.
"create" should mean it is creating a new setting.
"update" or "set" can be a function for editing an existing setting.

So we would have something like this:
if base_con_profile:
    wired_setting = wired.duplicate_setting(base_con_profile)
else:
     wired_setting = wired.create_setting(...) # No need for args if there are no "defaults".
if wired_setting:
    wired.set_setting(wired_setting, speed=speed, mtu=mtu,...)

Reply:
AFAIU the `else` is wrong because the `base_con_profile` might not contain a `wired_setting` (at least in theory, maybe it will always have one). IMHO the decision whether or not to create a setting would be better in the module to make it more clear for later refactoring in an interface to know what to do.